### PR TITLE
add Kiosk support

### PR DESCRIPTION
**Pull request by theScrabi**
### Add "Kiosk" function

A kiosk can be described as a page that provides content that you don't have to search for, like "top 50" or "trending".

This Pull Request:
* Supports Multiple Kiosks for one Service
* Add "trending" for Youtube
* Add "Top 50" and "New & Hot" for Soundcloud

The Idea is that with this we can finally get rid of the blank startpage of NewPipe.

### CODE REVIEW

**Comment by mauriciocolli** [comment]
I had a quick look at it and there's some points to consider:
- Hardcoded type will make it hard to translate it
- Isn't in sync with the latest codebase (using old Json dependency, for example)

**Comment by theScrabi** [comment]
We should fix this asap.

**Comment by theScrabi** [comment]
@mauriciocolli @wb9688 I resolved the collision.

**Comment by theScrabi** [comment]
Are we ready to merge?

**Comment by wb9688** [comment]
@theScrabi: What about:

>Hardcoded type will make it hard to translate it

**Comment by mauriciocolli** [comment]
There are some work to do here, but ok...

**PS:** I think you should have squashed these commits, they messed with the commit history with some useless commits. (instead of clicking `Merge pull request` use the `Squash and merge` option)

File: src/main/java/org/schabi/newpipe/extractor/services/soundcloud/SoundcloudStreamInfoItemExtractor.java [hunk 1]
```diff
@@ -30,12 +30,13 @@ public long getDuration() {
 
     @Override
     public String getUploaderName() {
-        return searchResult.getObject("user").getString("username");
+        //return searchResult.getObject("user").getString("username");
+        return searchResult.getObject("track").getObject("user").getString("username");
```
Comment (mauriciocolli) [inline src/main/java/org/schabi/newpipe/extractor/services/soundcloud/SoundcloudStreamInfoItemExtractor.java]: This change didn't considered that both charts and search (and others) use it, therefore, breaking some things.

Comment (mauriciocolli) [reply src/main/java/org/schabi/newpipe/extractor/services/soundcloud/SoundcloudStreamInfoItemExtractor.java]: Fixed in e9156ee499bec59a67f65ca6b78462cf40d026e9.

**Comment by theScrabi** [comment]
The type should be there to identify the different kiosks (like an Id) it is not there to be translated. We should do the translation on the client side, because otherwise wed have to find a way to make the extractor translatable.

File: src/main/java/org/schabi/newpipe/extractor/services/soundcloud/SoundcloudStreamInfoItemExtractor.java (hunk 1 above)
Comment (theScrabi) [reply src/main/java/org/schabi/newpipe/extractor/services/soundcloud/SoundcloudStreamInfoItemExtractor.java]: thanks :)

**Pull request merged**